gravitee-apim-e2e/api-test/portal/portal-applications.spec.ts

```diff
@@ -0,0 +1,90 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { afterAll, beforeAll, describe } from '@jest/globals';
+import { ANONYMOUS, forPortal, forPortalAsAdminUser, forPortalAsApiUser, forPortalAsAppUser, forPortalAsSimpleUser } from '@client-conf/*';
+import { created, noContent, unauthorized } from '../../lib/jest-utils';
+import { ApplicationApi } from '@portal-apis/ApplicationApi';
+import { Application } from '@portal-models/Application';
+import { PortalApplicationFaker } from '@portal-fakers/PortalApplicationFaker';
+
+const portalApplicationApiAsAdminUser = new ApplicationApi(forPortalAsAdminUser());
+const portalApplicationApiAsAppUser = new ApplicationApi(forPortalAsAppUser());
+const portalApplicationApiAsApiUser = new ApplicationApi(forPortalAsApiUser());
+const portalApplicationApiAsSimpleUser = new ApplicationApi(forPortalAsSimpleUser());
+const portalApplicationApiAsAnonymous = new ApplicationApi(forPortal({ auth: ANONYMOUS }));
+
+describe('Portal application tests with authorized users', () => {
+  let application: Application;
+
+  describe.each`
+    user             | applicationResource
+    ${'ADMIN'}       | ${portalApplicationApiAsAdminUser}
+    ${'api_user'}    | ${portalApplicationApiAsApiUser}
+    ${'app_user'}    | ${portalApplicationApiAsAppUser}
+    ${'simple_user'} | ${portalApplicationApiAsSimpleUser}
+  `('As $user user', ({ applicationResource }) => {
+    test('should create an application', async () => {
+      application = await created(
+        applicationResource.createApplicationRaw({
+          applicationInput: PortalApplicationFaker.newApplicationInput(),
+        }),
+      );
+    });
+
+    test('should delete an application', async () => {
+      await noContent(
+        applicationResource.deleteApplicationByApplicationIdRaw({
+          applicationId: application.id,
+        }),
+      );
+    });
+  });
+});
+
+describe('Portal application tests with unauthorized users', () => {
+  let application: Application;
+
+  beforeAll(async () => {
+    application = await portalApplicationApiAsAdminUser.createApplication({
+      applicationInput: PortalApplicationFaker.newApplicationInput(),
+    });
+  });
+
+  test('should fail to create an application as an unauthorized user', async () => {
+    await unauthorized(
+      portalApplicationApiAsAnonymous.createApplicationRaw({
+        applicationInput: PortalApplicationFaker.newApplicationInput(),
+      }),
+    );
+  });
+
+  test('should fail to delete an application as an unauthorized user', async () => {
+    await unauthorized(
```

Comment:
Should it be `unauthorized(...)` instead of `unauthorized(noContent(...))` ?

Reply:
Oops, that's true!